### PR TITLE
<ColorInput/> - drivers & test changes

### PR DESCRIPTION
**Pull request by mosheKerbel**
PR main purpose is for `<ColorInput/>` to be supported in React 16.
Currently many tests fail for this component in React 16.
As part of the required changes i have touched other areas.

Following have been changed:
1. Datahooks extraction
2. `Size` prop - support `medium` in `<input/>` instead of converting `medium` to `normal`.
3. UniDriver refactor - get rid of the `...` of nested components. "Breaking changes" are accepted here because only 1 consumer is using this testkit. These changes does not brake him.
4. Currently auto teskit docs generation produces a blank page. https://github.com/wix/wix-style-react/pull/3863 arijus will check it. In the meanwhile i am using a manual testkit readme so i had to modify it.

@mykas @shlomitc

### CODE REVIEW

File: src/ColorInput/ColorInput.js [hunk 1]
```diff
@@ -85,9 +85,10 @@ class ColorInput extends React.Component {
   static getDerivedStateFromProps(props, state) {
     if (!state.active && props.value !== state.value) {
       return {
-        ...state,
         value: extractHex(props.value),
       };
+    } else {
+      return null;
```
Comment (mykas) [inline src/ColorInput/ColorInput.js]: No need for else. Just do `return;`

File: src/Input/Input.driver.js [hunk 2]
```diff
@@ -103,6 +103,7 @@ const inputDriverFactory = ({ element }) => {
     isDisabled: () => element.classList.contains(styles.disabled),
     isOfStyle: style => element.classList.contains(styles[`theme-${style}`]),
     isOfSize: size => element.classList.contains(styles[`size-${size}`]),
+    getSize: () => element.getAttribute('data-size'),
```
Comment (mykas) [inline src/Input/Input.driver.js]: I think it is not a great idea to mix two components changes in one pr. In this pr you have added additional size `medium` for Input and also some new testkit method. This should be documented in the changelog later I'd say extract these changes and do another pr just for this.

Comment (mosheKerbel) [reply src/Input/Input.driver.js]: will do

File: src/ColorInput/ColorInput.js (hunk 1 above)
Comment (mosheKerbel) [reply src/ColorInput/ColorInput.js]: can get rid of the `else`, though i prefer  explicit `return null` over `return`

Comment (mykas) [reply src/ColorInput/ColorInput.js]: return; just returns undefined and in javascript we avoid `null` values. Be aware of these kind of issues: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null#Difference_between_null_and_undefined

Comment (mykas) [reply src/ColorInput/ColorInput.js]: For example a place like this: https://github.com/wix/wix-style-react/blob/master/src/ColorInput/components/ColorViewer.js#L14 could go very wrong if null values comes in :)

Comment (mosheKerbel) [reply src/ColorInput/ColorInput.js]: `Warning: %s.getDerivedStateFromProps(): A valid state object (or null) must be returned. You have returned undefined.,ColorInput`
so it's either `return null` or `return {}`
`return {}` is fine with me

Comment (mykas) [reply src/ColorInput/ColorInput.js]: if it requires an object then sure.

File: src/Input/Input.driver.js (hunk 2 above)
Comment (mosheKerbel) [reply src/Input/Input.driver.js]: https://github.com/wix/wix-style-react/pull/3891

**Comment by mosheKerbel** [comment]
failed due to unrelated flaky tests
merged

**Pull request merged**